File: 7.Reverse Integer/reverse_int.py
def reverse(x):
    """
    :type x: int
    :rtype: int
    """
    sign = [-1,1][x>0]
    ans = sign * int(str(abs(x))[::-1])      
        
    return (abs(ans)<((1<<31)-1)) and ans or 0

if __name__ == "__main__":
    print(reverse(-123456))

File: 7.Reverse Integer/reverse_int.c
#include <stdio.h>
#include <stdlib.h>

int reverse(int x){
    unsigned long long ans = 0;
    
    while (x != 0){
        ans = ans * 10 + x % 10;
        x /= 10;
    }
    
    return (ans < INT_MAX || ans > INT_MIN) ? ans : 0;
}

int main(){
    printf("%d\n", reverse(-12345678));
    return 0;
}